exporter/elasticsearchexporter/internal/objmodel/objmodel.go

```diff
@@ -424,7 +429,16 @@ func (doc *Document) iterJSONDedot(w *json.Visitor, otel bool) error {
 func StringValue(str string) Value { return Value{kind: KindString, str: str} }
 
 // IntValue creates a new value from an integer.
-func IntValue(i int64) Value { return Value{kind: KindInt, primitive: uint64(i)} }
+func IntValue(i int64) Value {
+	var v uint64
+	if i > 0 {
+		v = uint64(i) //nolint:gosec // overflow checked
+	}
+	return Value{kind: KindInt, primitive: v}
+}
```

Comment:
This looks strange. What if a negative integer is passed to `IntValue`? Wouldn't it be ignored?

We might as well just use an int64 as type of `primitive`.

Reply:
We need both `int64` and `uint64` depending on the cases. So if we switch `primitive` to `uint`, we'll just move the issue.

Reply:
I think df70c7e75a640f576908b0e021f2bdab7dd5ca72 is what you're looking for.